src/Http/Http.Results/src/ChallengeHttpResult.cs

```diff
@@ -6,84 +6,91 @@
     /// specified <paramref name="properties"/>.
     /// </summary>
     /// <param name="properties"><see cref="AuthenticationProperties"/> used to perform the authentication
     /// challenge.</param>
-    public ChallengeResult(AuthenticationProperties? properties)
+    public ChallengeHttpResult(AuthenticationProperties? properties)
         : this(Array.Empty<string>(), properties)
     {
     }
 
     /// <summary>
-    /// Initializes a new instance of <see cref="ChallengeResult"/> with the
+    /// Initializes a new instance of <see cref="ChallengeHttpResult"/> with the
     /// specified authentication scheme and <paramref name="properties"/>.
     /// </summary>
     /// <param name="authenticationScheme">The authentication schemes to challenge.</param>
     /// <param name="properties"><see cref="AuthenticationProperties"/> used to perform the authentication
     /// challenge.</param>
-    public ChallengeResult(string authenticationScheme, AuthenticationProperties? properties)
+    public ChallengeHttpResult(string authenticationScheme, AuthenticationProperties? properties)
         : this(new[] { authenticationScheme }, properties)
     {
     }
 
     /// <summary>
-    /// Initializes a new instance of <see cref="ChallengeResult"/> with the
+    /// Initializes a new instance of <see cref="ChallengeHttpResult"/> with the
     /// specified authentication schemes and <paramref name="properties"/>.
     /// </summary>
     /// <param name="authenticationSchemes">The authentication scheme to challenge.</param>
     /// <param name="properties"><see cref="AuthenticationProperties"/> used to perform the authentication
     /// challenge.</param>
-    public ChallengeResult(IList<string> authenticationSchemes, AuthenticationProperties? properties)
+    public ChallengeHttpResult(IList<string> authenticationSchemes, AuthenticationProperties? properties)
     {
-        AuthenticationSchemes = authenticationSchemes;
+        AuthenticationSchemes = authenticationSchemes.AsReadOnly();
         Properties = properties;
     }
 
-    public IList<string> AuthenticationSchemes { get; init; } = Array.Empty<string>();
+    /// <summary>
+    /// Gets the authentication schemes that are challenged.
+    /// </summary>
+    public IReadOnlyList<string> AuthenticationSchemes { get; internal init; } = Array.Empty<string>();
 
-    public AuthenticationProperties? Properties { get; init; }
+    /// <summary>
+    /// Gets the <see cref="AuthenticationProperties"/> used to perform the sign-out operation.
+    /// </summary>
+    public AuthenticationProperties? Properties { get; internal init; }
 
+    /// <inheritdoc/>
     public async Task ExecuteAsync(HttpContext httpContext)
     {
-        var logger = httpContext.RequestServices.GetRequiredService<ILogger<ChallengeResult>>();
+        var logger = httpContext.RequestServices.GetRequiredService<ILogger<ChallengeHttpResult>>();
```

Comment:
Do we care about changing log categories? I could see that being considered a breaking change. I don't see a huge benefit of changing it other than slightly cleaner code. @BrennanConroy @davidfowl

Reply:
I don't think this is super important for preview3, but I'd like to clarify our stance on changing log categories in general.

Reply:
Changing categories is considered a breaking change, example
https://github.com/dotnet/aspnetcore/blob/49270eeadd9830c0f8f0ffda5fcd02754da518a7/src/SignalR/common/Http.Connections/src/Internal/Transports/WebSocketsServerTransport.cs#L41-L42

Reply:
So, i think we should not change the categories. I have pushed a commit following @BrennanConroy example.

@halter73 / @BrennanConroy can you do a quick review? https://github.com/dotnet/aspnetcore/pull/40704/commits/00877b51023a5406e65df27e60431e49598081a5